packages/client/src/app/pages/forms/edit/builder/tabs/layout-editor/property-editor/form-controls/options-editor/custom-options.tsx

```diff
@@ -0,0 +1,292 @@
+import React, { useEffect, useState } from 'react';
+import {
+  AddOptionButton,
+  CheckboxWrapper,
+  Column,
+  DeleteButton,
+  DeleteIcon,
+  DragButton,
+  DragIcon,
+  H3,
+  Input,
+  PlusIcon,
+  Row,
+  TableAddOption,
+  TableOptions,
+  Wrapper,
+} from '@editor/builder/tabs/layout-editor/property-editor/form-controls/options-editor/custom-options.styles';
+import { CheckboxInput } from '@ff-client/app/components/form-controls/inputs/checkbox-input';
+import type { GenericValue } from '@ff-client/types/properties';
+
+type OptionProp = {
+  label: string;
+  value: string;
+  default: boolean;
+  autofocus?: boolean;
+};
+
+type PayloadProp = {
+  index: number;
+  key: string;
+  value: string | boolean;
+};
+
+export type CustomOptionsProps = {
+  useCustomValues: boolean;
+  options: OptionProp[] | GenericValue[];
+};
+
+type Props = {
+  handle: string;
+  value: CustomOptionsProps | GenericValue;
+  onChange?: (customOptions: CustomOptionsProps) => void;
```

Comment:
This will always be defined, so there's absolutely no necessity for complicating the codebase with `if (onChange)` calls.

```suggestion
  onChange: (customOptions: CustomOptionsProps) => void;
```